pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1810,18 +1800,23 @@ func (configgen *ConfigGeneratorImpl) onVirtualOutboundListener(
 		FilterChains: make([]plugin.FilterChain, len(ipTablesListener.FilterChains)),
 	}
 
+	// Set the protocol for each filter chain
+	for _, fc := range mutable.FilterChains {
+		fc.ListenerProtocol = plugin.ListenerProtocolTCP
```

Comment:
This will not take effect, fc is a value. I think you mean 

mutable.FilterChains[i].ListenerProtocol = plugin.ListenerProtocolTCP